### DISCUSSION TITLE
The New "Hoarding" Mechanic

### DISCUSSION THREAD

**Discussion by ShiningMinions** [sunflower-land/sunflower-land#1307] [General]
In my opinion, this new hoarding mechanic adds way too much friction towards players. I would have preferred to have a dialog between players/devs before this feature was added to game.

![hoarder](https://user-images.githubusercontent.com/103600068/182886571-03bdcac8-c996-4f75-aec1-bb4f4b8b3470.png)

As a player, it is extremely frustrating when I get this "Hoarding" message. I'm being forced to sync, and I cannot close this modal, nor undo my previous action. Here are my current choices:
1. I can hard-refresh my browser and lose all of my progress back to the last auto-save.
2. I can press the "Sync" button, but it's a pain because I would have preferred to do a few more actions before syncing (i.e. chop some more wood, harvest a few more crops, etc.) - then sync after these. A modal I cannot close stops me from this preparation.

How it should work instead:
1. Allow me to close the modal, which therefore reverts my previous action (i.e. the action which caused this modal to appear). This gives me a chance to prepare the rest of my farm for a sync.

Depending on the dialog of this Github Discussion, I am also willing to submit a "Change Request" Issue.

**Comment by adamhannigan** [comment]
@ShiningMinions Some great feedback 👌 

This makes sense to me. Do you want to have a go at implementing this?

**Reply by ShiningMinions** [reply]
Sounds great - I will try to get to this, but If I can't, I will submit a [CHANGE] Request and ask if another dev can pick it up :+1:

**Reply by Polysemouse** [reply]
Instead of a non-closable modal, we can show a confirmation dialog instead:

Eg. assume users need to sync when they collect 60 eggs 
1 . User harvests their 59th egg.  The harvest is successful 
2 . User attempts to harvest their 60th egg.  The game stops the user from harvesting, but instead shows a confirmation modal:

Are you a egg hoarder?
...(something along the lines of if you collect more egg you will need to sync and continue)...
[Back] [Sync]

3a . User clicks "Back" and closes the modal.  The 60th egg is not harvested.  Attempting to harvest the 60th egg will show that modal again.
3b . User clicks "Sync" and syncs the game

**Comment by ShiningMinions** [comment]
I added change-request here: https://github.com/sunflower-land/sunflower-land/issues/1354
